### ISSUE TITLE
How to avoid APNs DoS-attack blocking using node-apn

### ISSUE THREAD

**Issue by GuacemoleFour** [node-apn/node-apn#607]
Hi.

In Apples documentation called "Communicating with APNs", it is written that in order to prevent denial-of-service attack:

> You should leave a connection open unless you know it will be idle for an extended period of time—for example, if you only send notifications to your users once a day, it is acceptable practice to use a new connection each day.

Node-apn documentation state that node-apn:

> Maintains a connection to the server to maximise notification batching and throughput.

Here I'm uncertain what this means in practise. The original plan was to simply use Windows task scheduler for running the js-script every 15 minutes, checking the database for new notifications to send. When done sending notifications, it would run a ``` process.exit() ```
I do suspect that the server then would be reconnecting to APNs every 15 minutes, and we might get blocked by Apple. The new plan is to use a setTimeout-function, and keep the code responsible for reconnecting to APNs outside of the function.

My questions are (finally) as follows: 
1. Is my assumption true that we can not end the process every 15 minutes and keep the same connection?
2. If yes: What parts of the example code below is responsible for re-creating the connection with APNs, and should be kept out of the setTimeout-function running every 15 minutes?

```
//Load in the module
var apn = require('apn');

//Create a new connection to the Apple Push Notification provider API, passing a dictionary of options to the constructor.
var options = {
  token: {
  key: "...",
  keyId: "...",
  teamId: "..."
  },
  production: true
};
var apnProvider = new apn.Provider(options);

//Here follows code for connecting to database and check for new notifications to send:
//...
//...

//To send a notification you will first need a device token from your app as a string
let deviceToken = "a7ae07f4f6f55e659ba682b300d3833e5cf6539b64dc5a49200f26b9ff6a840d"

//Create a notification object, configuring it with the relevant parameters (See the notification documentation for more details.)
var note = new apn.Notification();

note.expiry = Math.floor(Date.now() / 1000) + 3600; 
note.sound = "ping.aiff";
note.alert = "Alert";
note.payload = {'messageFrom': 'John Appleseed'};
note.payload = {};
note.topic = "...";

//Send the notification to the API with send, which returns a promise.
apnProvider.send(note, deviceToken).then( (result) => {
console.log("Result from APNs: " + JSON.stringify(result, null, 2));   
    process.exit()
});
```

**Comment by florianreinhart** [comment]
Your assumption is correct.

You need to make sure that your `Provider` instance is created only once, i.e. outside of `setTimeout()`.
